skycoin/skywire
# Fix dmsg tracker

Did you run `make format && make check`?
yes

Fixes #1231 
Fixes #1233 

 Changes:	
- Removed dmsgtracker from hypervisor
- Added dmsgtracker as a module
- Updated msgTrackerManager

How to test this PR:
1. Checkout [#dmsg-175](https://github.com/skycoin/dmsg/pull/175)
2. In skywire uncomment `// replace github.com/skycoin/dmsg => ../dmsg` in `go.mod`
3. Run `make dep`
4. In `skywire-services` start a [integration env](https://github.com/SkycoinPro/skywire-services/blob/develop/docs/InteractiveEnvironments.md)
5. Let everything run for a few seconds and close the `dmsg-server` with `Ctrl+Z`
6. Check logs of visor B and UI
7. Stop the `dmsg-server` by getting it's PID from `htop` and killing it with `sudo kill -9 <pid>`
8. Check logs of visor B and UI
9. Re-start the `dmsg-server` with `source integration/dmsg-server-env.sh; ./bin/dmsg-server ./integration/dmsg-server.json --syslog localhost:514 --tag MSG_SRV`
10. Check the visor B
11. It should reconnect to the server
